### PR TITLE
Adapt doc strings and HowTo to mention progress bar

### PR DESCRIPTION
**Pull request by Astro-Kirsty**
**Description**
This PR is to resolve [#4679](https://github.com/gammapy/gammapy/issues/4679). 

I have added to the HowTo to list which methods contain the option to display the progress bar along with adding a small note in each of the docstring "The progress bar can be displayed for this function.".

This was my opinion of how to display these things, open to feedback on this.

### CODE REVIEW

**Comment by cgalelli** [comment]
Hi, Kirsty thanks for the implementation. In fact, as you mention in the linked issue, I also was not aware that some of these functions had a progress bar possibility. A tiny note about the docstring: Wouldn't it be better to highlight a bit more the possibility - for example by putting it in a separate "note" section on the same level as "parameters" - and perhaps linking the docs for the progress bar itself?

**Review by bkhelifi — APPROVED** [review]
Thanks a lot @Astro-Kirsty

**Review by registerrier — APPROVED** [review]
Thanks @Astro-Kirsty .

My main suggestion here would be to put a:
```
Note
----
```
and an indentation before the comments

**Review by bkhelifi — DISMISSED** [review]
Thanks @Astro-Kirsty

**Review by cgalelli — APPROVED** [review]
Everything is very readable now. Thanks!

File: gammapy/analysis/core.py [hunk 1]
```diff
@@ -156,7 +156,13 @@ def get_observations(self):
             log.debug(obs)
 
     def get_datasets(self):
-        """Produce reduced datasets."""
+        """
+        Produce reduced datasets.
+
+        Note
+        ----
+            The progress bar can be displayed for this function.
```
Comment (registerrier) [inline gammapy/analysis/core.py:164]:
```suggestion
        Note
        ----
        The progress bar can be displayed for this function.
```

File: gammapy/data/data_store.py [hunk 2]
```diff
@@ -347,6 +347,10 @@ def get_observations(
     ):
         """Generate a `~gammapy.data.Observations`.
 
+        Note
+        ----
+            The progress bar can be displayed for this function.
```
Comment (registerrier) [inline gammapy/data/data_store.py:352]:
```suggestion
        Note
        ----
        The progress bar can be displayed for this function.
```

File: gammapy/estimators/map/asmooth.py [hunk 3]
```diff
@@ -154,6 +154,10 @@ def _sqrt_ts_cube(cubes, method):
     def run(self, dataset):
         """Run adaptive smoothing on input MapDataset.
 
+        Note
+        ----
+            The progress bar can be displayed for this function.
```
Comment (registerrier) [inline gammapy/estimators/map/asmooth.py:159]:
```suggestion
        Note
        ----
        The progress bar can be displayed for this function.
```

File: gammapy/estimators/map/ts.py [hunk 4]
```diff
@@ -457,6 +457,10 @@ def run(self, dataset):
         Requires a MapDataset with counts, exposure and background_model
         properly set to run.
 
+        Note
+        ----
+            The progress bar can be displayed for this function.
```
Comment (registerrier) [inline gammapy/estimators/map/ts.py:462]:
```suggestion
        Note
        ----
        The progress bar can be displayed for this function.
```

File: gammapy/estimators/points/lightcurve.py [hunk 5]
```diff
@@ -98,6 +98,10 @@ def run(self, datasets):
 
         Normalize integral and energy flux between emin and emax.
 
+        Note
+        ----
+            The progress bar can be displayed for this function.
```
Comment (registerrier) [inline gammapy/estimators/points/lightcurve.py:103]:
```suggestion
        Note
        ----
        The progress bar can be displayed for this function.
```

File: gammapy/modeling/fit.py [hunk 6]
```diff
@@ -358,6 +358,10 @@ def stat_profile(self, datasets, parameter, reoptimize=False):
         The method used is to vary one parameter, keeping all others fixed.
         So this is taking a "slice" or "scan" of the fit statistic.
 
+        Note
+        ----
+            The progress bar can be displayed for this function.
```
Comment (registerrier) [inline gammapy/modeling/fit.py:363]:
```suggestion
        Note
        ----
        The progress bar can be displayed for this function.
```

File: gammapy/modeling/fit.py [hunk 7]
```diff
@@ -412,6 +416,10 @@ def stat_surface(self, datasets, x, y, reoptimize=False):
 
         See also: `Fit.stat_contour`
 
+        Note
+        ----
+            The progress bar can be displayed for this function.
```
Comment (registerrier) [inline gammapy/modeling/fit.py:421]:
```suggestion
        Note
        ----
        The progress bar can be displayed for this function.
```

File: gammapy/scripts/download.py [hunk 8]
```diff
@@ -64,6 +64,11 @@ def datasets_url(self):
 
 
 def progress_download(source, destination):
+    """
+    Note
+    ----
+        The progress bar can be displayed for this function.
```
Comment (registerrier) [inline gammapy/scripts/download.py:70]:
```suggestion
    Note
    ----
    The progress bar can be displayed for this function.
```

File: gammapy/utils/parallel.py [hunk 9]
```diff
@@ -115,6 +115,10 @@ def run_multiprocessing(
 ):
     """Run function in a loop or in Parallel
 
+    Note
+    ----
+        The progress bar can be displayed for this function.
```
Comment (registerrier) [inline gammapy/utils/parallel.py:120]:
```suggestion
    Note
    ----
    The progress bar can be displayed for this function.
```

**Review by registerrier — APPROVED** [review]
Thanks @Astro-Kirsty . I finally removed the indent as it is in fact ot required. Sorry for that.

**Pull request merged**